test/conformance/helpers/broker_tracing_test_helper.go

```diff
@@ -0,0 +1,489 @@
+	event := &resources.CloudEvent{
+		ID:       eventID,
+		Source:   senderName,
+		Type:     etTransformer,
+		Data:     fmt.Sprintf(`{"msg":%q}`, body),
+		Encoding: resources.CloudEventEncodingBinary,
+	}
+
+	// Send the CloudEvent (either with or without tracing inside the SendEvents Pod).
+	sendEvent := client.SendFakeEventToAddressable
+	if incomingTraceId {
+		sendEvent = client.SendFakeEventWithTracingToAddressable
+	}
+	if err := sendEvent(senderName, brokerName, common.BrokerTypeMeta, event); err != nil {
+		t.Fatalf("Failed to send fake CloudEvent to the broker %q", brokerName)
+	}
+
+	// TODO Actually determine the cluster's domain, similar to knative.dev/pkg/network/domain.go.
+	domain := "cluster.local"
+
+	// We expect the following spans:
+	// 0. Artificial root span.
+	// 1. Send pod sends event to the Broker Ingress (only if the sending pod generates a span).
+	// 2. Broker Ingress receives the event from the sending pod.
+	// 3. Broker Ingress sends the event to the Broker's TrChannel (trigger channel).
+	// 4. Broker TrChannel receives the event from the Broker Ingress.
+	// 5. Broker TrChannel sends the event to the Broker Filter for the "logger" trigger.
+	//     6. Broker Filter for the "logger" trigger receives the event from the Broker TrChannel.
+	//        This does not pass the filter, so this 'branch' ends here.
+	// 7. Broker TrChannel sends the event to the Broker Filter for the "transformer" trigger.
+	// 8. Broker Filter for the "transformer" trigger receives the event from the Broker TrChannel.
+	// 9. Broker Filter for the "transformer" trigger sends the event to the transformer pod.
+	// 10. Transformer pod receives the event from the Broker Filter for the "transformer" trigger.
+	// 11. Broker Filter for the "transformer" sends the transformer pod's reply to the Broker
+	//     InChannel (Ingress Channel).
+	// 12. Broker InChannel receives the event from the Broker Filter for the "transformer" trigger.
+	// 13. Broker InChannel sends the event to the Broker Ingress.
+	// 14. Broker Ingress receives the event from the Broker InChannel.
+	// 15. Broker Ingress sends the event to the Broker's TrChannel.
+	// 16. Broker TrChannel receives the event from the Broker Ingress.
+	// 17. Broker TrChannel sends the event to the Broker Filter for the "transformer" trigger.
+	//     18. Broker Filter for the "transformer" trigger receives the event from the Broker
+	//        TrChannel. This does not pass the filter, so this 'branch' ends here.
+	// 19. Broker TrChannel sends the event to the Broker Filter for the "logger" trigger.
+	// 20. Broker Filter for the "logger" trigger receives the event from the Broker TrChannel.
+	// 21. Broker Filter for the "logger" trigger sends the event to the logger pod.
+	// 22. Logger pod receives the event from the Broker Filter for the "logger" trigger.
+
+	// Useful constants we will use below.
+	ingressHost := fmt.Sprintf("%s-broker.%s.svc.%s", brokerName, client.Namespace, domain)
```

Comment:
Can we define these template strings as constants, possibly in `conformance/constants.go`? Seems they can be reused in other tests.

Reply:
Done.